langchain4j-core/src/main/java/dev/langchain4j/internal/GsonJsonCodec.java

```diff
@@ -16,6 +16,7 @@ class GsonJsonCodec implements Json.JsonCodec {
 
     private static final Gson GSON = new GsonBuilder()
         .setPrettyPrinting()
+        .setObjectToNumberStrategy(ToNumberPolicy.LONG_OR_DOUBLE)
```

Comment:
same here

Reply:
@langchain4j https://github.com/langchain4j/langchain4j/pull/1394 is a dedicated PR for this